cardano-cli/src/Cardano/CLI/Types/Common.hs

```diff
@@ -663,3 +667,12 @@ data PotentiallyCheckedAnchor anchorType anchor
   -- ^ Whether to check the hash or not (CheckHash for checking or TrustHash for not checking)
   }
   deriving (Eq, Show)
+
+-- | Type used for serialization when printing the hash of a transaction
+-- after having submitted it.
+newtype TxIdSubmission = TxIdSubmission {txhash :: T.Text}
+  deriving (Show, Generic)
+
+instance FromJSON TxIdSubmission
```

Comment:
I've put the `FromJSON` instance too, in case independent Haskell tooling wants to use it.

Reply:
What's wrong with `FromJSON TxId`?

Reply:
@Jimbo4350: see https://github.com/IntersectMBO/cardano-cli/pull/925#discussion_r1837907102